source/common/http/filter/cors_filter.cc

```diff
@@ -0,0 +1,155 @@
+#include "common/http/filter/cors_filter.h"
+
+#include "envoy/http/codes.h"
+
+#include "common/common/enum_to_int.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+
+namespace Envoy {
+namespace Http {
+
+CorsFilter::CorsFilter() : is_cors_request_(false) {}
+
+FilterHeadersStatus CorsFilter::decodeHeaders(HeaderMap& headers, bool) {
+  origin_ = headers.Origin();
```

Comment:
Per my other comment about default init of the policy, IMO, the first thing we should do here is the current contents of initialize(), then, if not enabled, quickly continue.